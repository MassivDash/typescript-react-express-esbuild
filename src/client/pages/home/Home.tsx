import React from 'react';
import { Main } from '@components';

const Home: React.FunctionComponent = () => {
  return (
    <Main>
      <h1 className="text-center text-2xl text-indigo-900 w-full h-64">
        Esbuild only setup
      </h1>
      <h2 className="text-center text-lg w-full h-44">
        This boilerplate is experimental typescript, react, node.js (express)
        setup, completly dropping webpack family in favor of new Esbuild
        package. Project uses tailwindcss as the main css framework
      </h2>
    </Main>
  );
};

export default Home;
