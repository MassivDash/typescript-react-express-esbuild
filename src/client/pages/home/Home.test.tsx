import { render } from '@testing-library/react';
import React from 'react';
import Home from './Home';

describe('Home', () => {
  it('should render', () => {
    const { asFragment, queryByText } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
    expect(queryByText('Esbuild only setup')).toBeTruthy();
  });
});
