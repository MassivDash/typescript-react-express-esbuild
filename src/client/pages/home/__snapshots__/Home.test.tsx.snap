// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
<DocumentFragment>
  <main
    class="mt-10 mx-auto max-w-7xl px-4 sm:mt-12 sm:px-6 md:mt-16 lg:mt-20 lg:px-8 xl:mt-28"
  >
    <h1
      class="text-center text-2xl text-indigo-900 w-full h-64"
    >
      Esbuild only setup
    </h1>
    <h2
      class="text-center text-lg w-full h-44"
    >
      This boilerplate is experimental typescript, react, node.js (express) setup, completly dropping webpack family in favor of new Esbuild package. Project uses tailwindcss as the main css framework
    </h2>
  </main>
</DocumentFragment>
`;
